pkg/api/customization/node/node.go

```diff
@@ -31,4 +38,44 @@ func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
 	if nodeTemplateID == nil && customConfig == nil {
 		delete(resource.Links, "remove")
 	}
+
+	if convert.ToBool(resource.Values["unschedulable"]) {
+		resource.AddAction(apiContext, "uncordon")
+	} else {
+		resource.AddAction(apiContext, "cordon")
+	}
+}
+
+type ActionWrapper struct{}
+
+func (a ActionWrapper) ActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	switch actionName {
+	case "cordon":
+		return cordonUncordonNode(actionName, apiContext, true)
+
+	case "uncordon":
+		return cordonUncordonNode(actionName, apiContext, false)
+	}
+
+	return nil
+}
+
+func cordonUncordonNode(actionName string, apiContext *types.APIContext, cordon bool) error {
+	var node map[string]interface{}
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &node); err != nil {
+		return httperror.NewAPIError(httperror.InvalidReference, "Error accessing node")
+	}
+	schema := apiContext.Schemas.Schema(&managementschema.Version, client.NodeType)
+	unschedulable := convert.ToBool(values.GetValueN(node, "unschedulable"))
+	if cordon && unschedulable {
+		return httperror.NewAPIError(httperror.InvalidAction, fmt.Sprintf("Node %s already cordoned", apiContext.ID))
+	} else if !(cordon || unschedulable) {
+		return httperror.NewAPIError(httperror.InvalidAction, fmt.Sprintf("Node %s already schedulable", apiContext.ID))
```

Comment:
it feels like this message can be ambiguous based on cordon/unschedulable values. Can we actually throw some sort of generalized error in situation when `cordon == unschedulable` (it should cover both use cases)?